src/migrations/20230408000000-add-foreign-key-indices.js

```diff
@@ -0,0 +1,78 @@
+module.exports = {
+  up: (queryInterface) => queryInterface.sequelize.transaction((transaction) => Promise.all([
+    // Add foreign key indices.
+    // In link tables, add a compound index plus a single-column index for the second key in the
+    // compound index. The compound index works about as well as a single-column index for the
+    // first key even when the second is not referenced. In cases where there's a unique
+    // constraint, postgres already supplies an index so we only add the second key index.
+    // We're also not adding indexes on low-cardinality foreign keys like region, roleId,
+    // and topicId
+    queryInterface.addIndex('ActivityReportGoals', ['activityReportId', 'goalId'], { transaction }),
+    queryInterface.addIndex('ActivityReportGoals', ['goalId'], { transaction }),
+    queryInterface.addIndex('ActivityReportCollaborators', ['activityReportId'], { transaction }),
+    queryInterface.addIndex('ActivityReportObjectives', ['activityReportId', 'objectiveId'], { transaction }),
+    queryInterface.addIndex('ActivityReportObjectives', ['objectiveId'], { transaction }),
+    queryInterface.addIndex('ActivityReportObjectiveFiles', ['fileId'], { transaction }),
+    queryInterface.addIndex('Objectives', ['goalId'], { transaction }),
+    queryInterface.addIndex('Objectives', ['objectiveTemplateId'], { transaction }),
+    queryInterface.addIndex('ObjectiveFiles', ['objectiveId', 'fileId'], { transaction }),
+    queryInterface.addIndex('ObjectiveFiles', ['fileId'], { transaction }),
+    // Skipping creating a userProvidedUrl index because it's just a text field
+    queryInterface.addIndex('ObjectiveResources', ['objectiveId'], { transaction }),
+    queryInterface.addIndex('ObjectiveTopics', ['objectiveId'], { transaction }),
+    queryInterface.addIndex('ObjectiveTemplateFiles', ['objectiveTemplateId', 'fileId'], { transaction }),
+    queryInterface.addIndex('ObjectiveTemplateFiles', ['fileId'], { transaction }),
+    // Skipping creating a userProvidedUrl index because it's just a text field
+    queryInterface.addIndex('ObjectiveTemplateResources', ['objectiveTemplateId'], { transaction }),
+    queryInterface.addIndex('ObjectiveTemplateTopics', ['objectiveTemplateId'], { transaction }),
+    queryInterface.addIndex('ObjectiveTemplateRoles', ['objectiveTemplateId'], { transaction }),
+    queryInterface.addIndex('Goals', ['grantId'], { transaction }),
+    queryInterface.addIndex('Goals', ['goalTemplateId'], { transaction }),
+    queryInterface.addIndex('GoalTemplateObjectiveTemplates', ['goalTemplateId', 'objectiveTemplateId'], { transaction }),
+    queryInterface.addIndex('GoalTemplateObjectiveTemplates', ['objectiveTemplateId'], { transaction }),
+    queryInterface.addIndex('GroupGrants', ['groupId', 'grantId'], { transaction }),
+    queryInterface.addIndex('GroupGrants', ['grantId'], { transaction }),
```

Comment:
Could this list be generated via making indexes off all present foreign keys per table, factorial combination,  populate it as via a function. make a trigger for the function on table create or alter to automatically be added as the db changes

Reply:
I think we want to be more judicious in our addition of indexes off foreign keys, because sometimes they add more write overhead than it helps with reading. If the db was being used purely transactionally and never for charts, even fewer foreign keys would be useful. As it was, I actually removed half a dozen indices after reconsidering whether they would really help any actual query the system might perform.